airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -147,9 +145,20 @@ private void runInternal(final IntegrationConfig parsed) throws Exception {
           final JsonNode config = parseConfig(parsed.getConfigPath());
           validateConfig(integration.spec().getConnectionSpecification(), config, "WRITE");
           final ConfiguredAirbyteCatalog catalog = parseConfig(parsed.getCatalogPath(), ConfiguredAirbyteCatalog.class);
-          try (final AirbyteMessageConsumer consumer = destination.getConsumer(config, catalog, outputRecordCollector)) {
-            runConsumer(consumer);
-          }
+
+          final Procedure consumeWriteStreamCallable = () -> {
+            try (final SerializedAirbyteMessageConsumer consumer = destination.getSerializedMessageConsumer(config, catalog, outputRecordCollector)) {
```

Comment:
for understanding: after this merge, we should be able to get rid of `getConsumer`?

Reply:
we can only get rid of `getConsumer` once all destinations are migrated over to use the new thing. but for snowflake, yes, in subsequent PRs it overrides this method instead.